test/e2e/common/node/container_probe.go

```diff
@@ -246,6 +246,26 @@ var _ = SIGDescribe("Probing container", func() {
 		runReadinessFailTest(f, pod, time.Minute)
 	})
 
+	/*
+		Release: v1.21
+		Testname: Pod liveness probe, container exec timeout, restart
+		Description: A Pod is created with liveness probe with a Exec action on the Pod. If the liveness probe call does not return within the timeout specified, liveness probe MUST restart the Pod. When ExecProbeTimeout feature gate is disabled and cluster is using dockershim, the timeout is ignored BUT a failing liveness probe MUST restart the Pod.
+	*/
+	ginkgo.It("should be restarted with a failing exec liveness probe that took longer than the timeout", func() {
```

Comment:
@andrewsykim You asked why we aren't running this test in an ExecProbeTimeout=false scenario only. The main reason is that not all ExecProbeTimeout=false scenarios will pass this new test (e.g., containerd). I think the value of this new test is to guarantee that a failing probe fails *no matter what*. Since we have tests that validate timeouts fire as expected, this test should in practice be a reproduction of that pod restart condition (pod restarted due to timeout), but this additional test coverage will ensure that certain pre-1.20 behaviors (containerd, for example) will be caught right away in E2E tests if they resurface for any weird reason.